src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -118,6 +119,11 @@ class ProtobufPlugin implements Plugin<Project> {
           // block. Only at this point the configuration has been finalized.
           project.protobuf.tools.registerTaskDependencies(project.protobuf.getGenerateProtoTasks().all())
         }
+
+        project.plugins.withType(IdeaPlugin) { IdeaPlugin ip ->
+            ip.model.module.sourceDirs += project.file("${project.protobuf.generatedFilesBaseDir}/main/java");
+            ip.model.module.generatedSourceDirs += project.file("${project.protobuf.generatedFilesBaseDir}/main/java");
+        }
```

Comment:
Have you considered moving this logic into doApply(), after calling `project.protobuf.generateProtoTasks.all()*.doneConfig()`? At that point the sourceSets and builtin/plugin outputs are finalized, then you should be able to add all generated sourceSets automatically, and the user wouldn't need to do this for additional sourceSets/builtins/plugins in `build.gradle`.